neuraxle/higher_order_steps.py

```diff
@@ -0,0 +1,117 @@
+"""
+Neuraxle's Higher Order Steps
+===============================================================================================================
+A higher-order step is a function that takes an existing step and returns another step that wraps it.
+An higher-order step is never be saved because it is note serializable.
+It is useful for applying side effects that are outside of the scope of the pipeline execution.
+
+..
+    Copyright 2019, Neuraxio Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+"""
+from abc import ABC
+
+from neuraxle.base import BaseTransformer, ExecutionContext, MetaStep, ForceHandleMixin
+from neuraxle.data_container import DataContainer
+
+
+class BaseHigherOrderStep(ForceHandleMixin, MetaStep, ABC):
+    """
+    Base class to represent an higher order step.
+    A higher-order step is just a function that takes an existing step and returns another step that wraps it.
+
+    An higher order step:
+        - cannot be saved
+        - cannot modify the data container
+        - cannot push himself in the execution context
+
+    .. seealso::
+        :class:`~neuraxle.base.MetaStep`
+    """
+    def __init__(self, step: BaseTransformer):
+        MetaStep.__init__(self, wrapped=step)
+        ForceHandleMixin.__init__(self)
+
+    def save(self, context: ExecutionContext, full_dump=False):
+        """
+        Don't save the higher order step. Save the wrapped step only.
+
+        :param context: context to save.
+        :param full_dump: save full dump or not
+        :return: saved wrapped step
+        """
+        if len(context) > 0 and isinstance(context.parents[-1], BaseHigherOrderStep):
+            context.pop()
+        return self.wrapped.save(context, full_dump=full_dump)
```

Comment:
Please add a saver to `class BaseHigherOrderStep`. 

- **Upon Saving**: The saver will `del step.context` to strip the BaseHigherOrderStep of its context. 
- **Upon loading**: The new context that is available and that we have at hand should be passed and set into the BaseHigherOrderStep.